src/realm/util/logger.cpp

```diff
@@ -175,12 +175,25 @@ const std::string_view Logger::level_to_string(Level level) noexcept
     return "";
 }
 
+std::shared_ptr<std::mutex> StderrLogger::m_stderr_log_mutex = std::make_shared<std::mutex>();
+
+StderrLogger::StderrLogger() noexcept
+    : m_log_mutex(StderrLogger::m_stderr_log_mutex)
+{
+}
+
 void StderrLogger::do_log(const LogCategory& cat, Level level, const std::string& message)
 {
-    static Mutex mutex;
-    LockGuard l(mutex);
+    do_write(util::format("%1 - %2%3", cat.get_name(), get_level_prefix(level), message));
+}
+
+void StderrLogger::do_write(std::string&& output)
+{
+    REALM_ASSERT(m_log_mutex);
+    // Lock the mutex to avoid comingling the logger output messages
```

Comment:
something i've never understood is why we need this mutex at all. std::cerr *is* threadsafe on its own. if the C++ standard library is working correctly, you should never see comingling of logger output messages to stderr.

Reply:
Removed the mutex - a single `<<` call to `std::cerr` is thread safe, so output message is created first and then pushed to stderr as a single string.

Reply:
to be clear, i meant this as a question rather than a statement. is the reason we added the mutex here because we were seeing interleaving or something else?

Reply:
There was some interleaving in the past during the tests. For std::cerr, each individual streaming operator call is thread safe, but chained operators are not, allowing context changes to happen before the entire message is printed.
e.g., `std::cerr << "string a" << "string b";` ==> `std::cerr << "string a"; std::cerr << "string b";`

Reply:
But unfortunately, MacOS TSAN is not happy with writing to `std::cerr` from two threads, but it is fine with `fprintf()` and supposedly it is faster then `std::cerr`.  I updated `StderrLogger` to use `fprintf()` instead.